Publish minutes of 2023-06-08 meeting (#408)
hober/web-extensions: _minutes/2023-06-08-wecg.md

[Issue 404](https://github.com/w3c/webextensions/issues/404): Proposal: Allow notifications API in content scripts

 * [tomislav] Didn't see any use case that helps with this beyond not waking up the service worker. The API itself is already async, so the delay.
 * [rob] And also, the web already has a Notification API.
 * [oliver] Agreed.
 * [timothy] Agreed.
 * Consensus: Close request.

[Issue 407](https://github.com/w3c/webextensions/issues/407): Inconsistency: returned value for windows.create({incognito: true})

 * [simeon] Opening an incognito window in Chrome returns a null value, in Firefox it returns window data.
 * [rob] If an extension has access to incognito, opening a window should return the details of the window. If the extension does not have access to incognito, the method should reject the call and the caller should not see the window.
 * [david] In Safari 17 we updated the implementation, we offer the ability to restrict extensions in private browsing.
 * [timothy] Currently we don't even let you open a window if you're in private mode. We return null and don't open a window.
 * [oliver] Chrome returns null and does open the window.

>  * [rob] Do you return null or error? Seems wrong to return null when you don't perform the requested action. From an API design perspective, I think that a rejection with an error would be appropriate here.

Reviewer: ```suggestion
 * [rob] Does Safari return null or error? Seems wrong to return null when you don't perform the requested action. From an API design perspective, I think that a rejection with an error would be appropriate here.
```

 * [oliver] What's your thinking for returning information about the window if you can't interact with it?
 * [rob] Shouldn't provide information about the window if you can't interact with it.
 * [maxim] Note that this is specific to [incognito:split](https://developer.chrome.com/docs/extensions/mv3/manifest/incognito/#split) in Chrome.

>  * [timothy] Safari only supports incognito:spanning, not split.

Reviewer: @dotproto, does this sound right?

```suggestion
 * [simeon] Unclear what the behaviour in split is but this conversation is specifically around the behaviour in spanning when incognito access has not been granted.
 * [timothy] Safari only supports incognito:spanning, not split.
```

 * [rob] Firefox also only supports incognito:spanning, not split.
 * [rob] If there is a desire to hide incognito windows in split, the API should not allow extensions to create a private window from a non-private extension context. An extension should then communicate with the other part of its extension to intentionally interact with incognito contexts.

>  * [mukul] There may be use cases where an extension wants to open, say, a banking site in private mode, but otherwise doesn't want to interact.

Reviewer: Adding a missing part of the conversation (cc/ @zombie):

```suggestion
 * [oliver] Aren't there cases on the web where you can open a window but not interact with it?
 * [tomislav] An example would be target="_blank".
 * [mukul] There may be use cases where an extension wants to open, say, a banking site in private mode, but otherwise doesn't want to interact.
```
